### ISSUE TITLE
Dropbox login crash on Android 13 with targetSdk 33

### ISSUE THREAD

**Issue by DynamicG-Android** [dropbox/dropbox-sdk-java#406]
Hi,
I've lifted my app to targetSdk 33. The Dropbox login flow crashes on my Android 13 device when pressing "Allow" in the "grant access" dialog in the Dropbox app (i.e. throwing ActivityNotFound when trying to call back to my app). Very likely root cause: "Intent filters block non-matching intents" change introduced with Android 13 SDK 33. It works fine when reverting my app to targetSdk 31.

Are you aware of this issue? I assume this needs a patch on your side (?)

Stack trace:
```
Process: [com.dropbox.android](http://com.dropbox.android/)
    android.content.ActivityNotFoundException: Unable to find explicit activity class {com.myapp.MyActivity/com.dropbox.core.android.AuthActivity}; have you declared this activity in your AndroidManifest.xml, or does your intent not match its declared <intent-filter>?
        at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2158)
        at android.app.Instrumentation.execStartActivity(Instrumentation.java:1805)
        at android.app.Activity.startActivityForResult(Activity.java:5470)
        at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:2)
        at android.app.Activity.startActivityForResult(Activity.java:5428)
        at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:1)
        at android.app.Activity.startActivity(Activity.java:5926)
        at android.app.Activity.startActivity(Activity.java:5893)
        at dbxyzptlk.Wd.a.b(RealSafeIntentStarter.java:3)
        at com.dropbox.android.dauth.DelegatedAuthActivity$j$c.a(DelegatedAuthActivity.java:9)
        at com.dropbox.android.dauth.DelegatedAuthActivity$j.a(DelegatedAuthActivity.java:2)
        at dbxyzptlk.Vg.c.onPostExecute(DbxAsyncTask.java:10)
        at android.os.AsyncTask.finish(AsyncTask.java:771)
        at android.os.AsyncTask.-$$Nest$mfinish(Unknown Source:0)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:788)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7898)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)

```

**Comment by greg-db** [comment]
Thanks for the report! We'll look into it.

**Comment by devPalacio** [comment]
Hey @DynamicG-Android, could you show me what your manifest looks like?

Are you using the com.dropbox.core.android.AuthActivity like the example app? 

I got it working on the example app by adding the following intent filter under the AuthActivity
```         
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
```

Let me know if that helps.

**Comment by DynamicG-Android** [comment]
perfect. yes this works. many thanks. (and yes, my manifest entry is like in the example app).

recap for those stumbling over the same problem and seeing this - the second intent-filter needs to be added to manifest when targeting Android 13:
```
        <activity
            android:name="com.dropbox.core.android.AuthActivity"
            android:exported="true"
            android:configChanges="orientation|keyboard"
            android:launchMode="singleTask">
            <intent-filter>
                <!-- Change this to be db- followed by your app key -->
                <data android:scheme="db-${dropboxKey}" />
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.BROWSABLE" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
            <!-- for Android 13 -->
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
            </intent-filter>
        </activity>

```

**Comment by devPalacio** [comment]
Fixed. We've shipped some changes in our Dropbox application and the extra intent filter is no longer required. Feel free to remove it from your manifest.

**Comment by michaelhuber88** [comment]
@devPalacio thank you for fixing this. I assume it is save to leave the intent filter for users which haven't upgraded their dropbox app yet, right?

**Comment by devPalacio** [comment]
Yes, this is a valid as a workaround until you're confident your users are running a newer version of Dropbox.

**Comment by handstandsam** [comment]
FYI: This version @devPalacio mentions v`298.2.2` with the fix is available to all users starting September 20, 2022.

**Comment by handstandsam** [comment]
We've determined a fix from the Dropbox app side for this issue.  We needed to explicitly add in the `data` field to the `Intent` and add the `android.intent.category.BROWSABLE` starting with this change in Android 13.

I'm going to re-open this issue until the fix in the Dropbox application is widely available.

**WORKAROUND**: This issue can be avoided using the answer above https://github.com/dropbox/dropbox-sdk-java/issues/406#issuecomment-1215155110
